### PR TITLE
refactor(goal_planner): reduce call to isSafePath

### PR DESCRIPTION
**Pull request by soblin**
## Description

depends https://github.com/autowarefoundation/autoware.universe/pull/8795

## Related links

- [CompanyName internal link](https://tier4.atlassian.net/wiki/spaces/~621c200f9c3cce006949b60e/pages/3286696258/GoalPlanner#isSafePath%E3%81%AE%E5%91%BC%E3%81%B3%E5%87%BA%E3%81%97%E5%9B%9E%E6%95%B0)

## How was this PR tested?

https://evaluation.tier4.jp/evaluation/reports/98dbc36e-c79a-53f0-a930-5aac27de1c95?project_id=prd_jt

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp [hunk 1]
```diff
@@ -875,8 +883,15 @@ BehaviorModuleOutput GoalPlannerModule::plan()
 {
   universe_utils::ScopedTimeTrack st(__func__, *time_keeper_);
 
+  if (!context_data_) {
+    RCLCPP_ERROR(getLogger(), " [pull_over] plan() is called without valid context_data");
+  }
+  const auto & context_data = context_data_.has_value()
+                                ? context_data_.value()
+                                : PullOverContextData(std::make_pair(true, true));
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp:891]:
```suggestion
  const auto context_data = context_data_.has_value()
                                ? context_data_.value()
                                : PullOverContextData(std::make_pair(true, true));
```
temporary object should not be bound

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp [hunk 2]
```diff
@@ -276,11 +276,15 @@ void GoalPlannerModule::onTimer()
       RCLCPP_DEBUG(getLogger(), "has previous module path shape changed");
       return true;
     }
+    // TODO(soblin): never call isSafePath on thread pool side
+    const auto local_context_data = PullOverContextData(isSafePath(
+      local_planner_data, parameters, ego_predicted_path_params, objects_filtering_params,
+      safety_check_params));
     if (
       hasDeviatedFromLastPreviousModulePath(local_planner_data) &&
+      // TODO(soblin):
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp:285]: please writer the details of todo

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp [hunk 3]
```diff
@@ -1529,18 +1520,26 @@ void GoalPlannerModule::postProcess()
 {
   universe_utils::ScopedTimeTrack st(__func__, *time_keeper_);
 
+  if (!context_data_) {
+    RCLCPP_ERROR(getLogger(), " [pull_over] postProcess() is called without valid context_data");
+  }
+  const auto & context_data = context_data_.has_value()
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp:1526]:
```suggestion
  const auto context_data = context_data_.has_value()
```

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp [hunk 4]
```diff
@@ -1594,8 +1591,16 @@ BehaviorModuleOutput GoalPlannerModule::planWaitingApproval()
 {
   universe_utils::ScopedTimeTrack st(__func__, *time_keeper_);
 
+  if (!context_data_) {
+    RCLCPP_ERROR(
+      getLogger(), " [pull_over] planWaitingApproval() is called without valid context_data");
+  }
+  const auto & context_data = context_data_.has_value()
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp:1598]:
```suggestion
  const auto context_data = context_data_.has_value()
```

**Comment by soblin** [comment]
I applied suggestions

**Pull request merged**